### ISSUE TITLE
Occasional problem with height

### ISSUE THREAD

**Issue by pliablepixels** [arielfaur/ionic-pullup#1]
Hello, first off, thank you for this lovely plugin. I am facing an occasional problem with height calculation -- once in a while and very randomly, the pullup bar when tapped goes beyond the height of the content window. I have not specified a max-height and according to your code, it auto calculates the height -- what is confusing me is that it only happens once in a while - and I can't seem to predict when and I don't see an error

To try and fix, I forced a "-20" to your computation (see my version at  https://github.com/pliablepixels/zmNinja/blob/master/www/external/ion-pullup.js#L48)  - while that reduces the number of times it goes beyond the content-height it still happens. I'm quite confused why

Platform: iOS9, iphone 5

If it helps, here is how I am using your pullup
https://github.com/pliablepixels/zmNinja/blob/master/www/templates/events.html#L190

Thanks again for any insight.

**Comment by arielfaur** [comment]
Hi! I haven't been able to test on iOS9 yet, so cannot confirm this yet. My problem is I recently upgraded my iPhone 5 to iOS9 and I am now unable to deploy to iOS on Mavericks. I will need to update OSX and Xcode first :(
As soon as I do that I will run some test and get back to you.

**Comment by pliablepixels** [comment]
Okay, I found out why this is happening. Here is an easy way to replicate this:

a) Invoke the pullup in portrait mode
b) Rotate the screen to horizontal so the orientation change event occurs
c) switch app to background
d) bring  app to foreground
e) tap on pullup in portrait mode and you will see the bar goes beyond the content screen

The problem is that in your computation, headerHeight returns 0 in this case while it should return 64
So there is something going on while switching orientation and moving the app to background/foreground that is causing this problem,. I've hardcoded header height to 64 for now, but its not a good solution. 

I'll keep debugging - and try and figure out why headerHeight returns a 0 value in this situation.

**Comment by arielfaur** [comment]
You read my mind. I was about to write whether it was related to screen orientation. As you see in the code, I need to set a timeout before recomputing heights on screen rotation. This is needed because ```window.innerHeight``` is not updated immediately with the new values. We may need to increase the timeout to 1 sec.  That really sucks, but you can give it a try.

**Comment by pliablepixels** [comment]
@arielfaur the issue however is window.innerHeight is correct - it is headerHeight that returns 0

**Comment by arielfaur** [comment]
Ok, I finished the upgrade to Xcode 7. Now I can debug. Will get back to you later.

**Comment by arielfaur** [comment]
I cannot reproduce your issue. I am testing on iOS9 and it seems to work fine when I switch apps and rotate the device as you suggested. Have you found anything else?

**Comment by arielfaur** [comment]
One thing I'd like to fix is the flashing whenever the device is rotated, but that is unfortunately a WebKit bug as I need set a timeout to compute heights.

**Comment by pliablepixels** [comment]
That's odd. I can replicate that 100% of the time. I've hardcoded header = 64 (as my app always shows the header) and the problem has gone away, so it looks in that situation https://github.com/arielfaur/ionic-pullup/blob/master/dist/ion-pullup.js#L23 returns null so it takes the headerHeight at 0. 

When you say 'flashing' what do you mean? What I noticed is that when you rotate, the pullup does move around a bit and then collapses (does not stay open) but there is no flashing

**Comment by arielfaur** [comment]
Yeah, that's what I meant with flashing, sorry (it expands and then collapses to accommodate to the new screen). But that's the result of the ```computeHeights``` function and the timeout set.

I am trying once again to replicate your issue. So do you first rotate the device and then send the app to the background, or do you rotate it after bringing the app to the foreground? I followed your steps above but I don't see the issue.

**Comment by pliablepixels** [comment]
I created a video for you (its in 3 parts as Quick time does not allow for recording the phone when you rotate)

1. Using the app in portrait  mode - everything great
2. Switching app to landscape mode - everything great
3. Move to background and resume -- headerHeight becomes 0 and remains at 0 till I kill and restart

https://www.youtube.com/watch?v=QIn76JgQWZs&list=PLNjE3bH-2rZKOoFCz4csn8zYA7pmXNYfH

**Comment by arielfaur** [comment]
I swear I've tried every possible combination and still can't see the issue on my device. I am using the following parameters as you suggested:
```
initial-state="minimized" default-behavior="expand"
```
I wonder now what could cause different behavior on different devices. I am running iOS9 on and iPhone 5C.

**Comment by pliablepixels** [comment]
Okay, so I know the problem, but I don't know enough to fix it. Here is the analysis:

This is not an IOS9 problem. It happens for me in Android too (4.2, Galaxy S2). I think it depends on how the app is written and how heavy it is - you may not be seeing this with just the pullup code because everything is rendering within the right time for it to compute

Here is what is happening (and this is more visible in my android device which is much slower)

headerHeight is getting computed before ionic is rendering the header on the screen, so the code that retrieves the header class is returning null. 

I'll do some more debugging tomorrow and see how to fix it (if I can)

**Comment by arielfaur** [comment]
The moment you mentioned the Galaxy S2 I remembered I had one too somewhere around the house. I tested your use case and I can see the odd behavior as soon as you bring the app back to the foreground. I will try to find the reason why ```headerHeight``` has the wrong value.

**Comment by arielfaur** [comment]
@pliablepixels I think I've found a fix for this issue. Could you please check out this version?
https://github.com/arielfaur/ionic-pullup/blob/fix-orientation-issue/dist/ion-pullup.js

Let me know if this version solves the issue. I've tested on my S2 and iPhone 5.  I have also reduced the timeout to 300 ms for a better user experience. Wish I could get rid of it completely but it won't work.

**Comment by pliablepixels** [comment]
@arielfaur unfortunately this does not fix it. I've fixed the problem but it looks like your repo is not accepting Pull requests

Here is the summary of the problem: I've implemented pin authentication for my app. Any time the app resumes from the background, I use ``$state.go`` to force the user to go to a state authentication view and go back to the previous view. The state authentication view does not have a header. While this transition is happening, it looks like your directive computes height and that creates a problem. This may happen in other apps too.

To fix this I moved the computation code to onExpand() so its done at the right time. If you would like to enable accepting pull requests I can push it to you for review, else I can upload it to a gist
